Examples/CaseStudies/SwiftUICaseStudiesTests/01-GettingStarted-SharedStateTests.swift

```diff
@@ -0,0 +1,77 @@
+import Combine
+import ComposableArchitecture
+import XCTest
+
+@testable import SwiftUICaseStudies
+
+class SharedStateTests: XCTestCase {
+  func testTabRestoredOnReset() {
+    let store = TestStore(
+      initialState: SharedState(),
+      reducer: sharedStateReducer,
+      environment: ()
+    )
+    
+    store.assert(
+      .send(.selectTab(.profile)) {
+        $0.currentTab = .profile
+      },
+      .send(.profile(.resetCounterButtonTapped)) {
+        $0.currentTab = .counter
+      })
+  }
+
+  func testTabSelection() {
+    let store = TestStore(
+      initialState: SharedState(),
+      reducer: sharedStateReducer,
+      environment: ()
+    )
+
+    store.assert(
+      .send(.selectTab(.profile)) {
+        $0.currentTab = .profile
+      },
+      .send(.selectTab(.counter)) {
+        $0.currentTab = .counter
+      })
+  }
+
+  func testIsPrimeWhenPrime() {
+    let store = TestStore(
+      initialState: SharedState.CounterState(alert: nil, count: 3, maxCount: 0, minCount: 0, numberOfCounts: 0),
+      reducer: sharedStateCounterReducer,
+      environment: ()
+    )
+
+    store.assert(
+      .send(.isPrimeButtonTapped) {
+        $0.alert = .init(
+          title: "👍 The number \($0.count) is prime!"
+        )
+      },
+      .send(.alertDismissed) {
+        $0.alert = nil
+      }
+    )
+  }
+
+  func testIsPrimeWhenNotPrime() {
+    let store = TestStore(
+      initialState: SharedState.CounterState(alert: nil, count: 6, maxCount: 0, minCount: 0, numberOfCounts: 0),
+      reducer: sharedStateCounterReducer,
+      environment: ()
+    )
+
+    store.assert(
+      .send(.isPrimeButtonTapped) {
+        $0.alert = .init(
```

Comment:
Another case where it might be nice to have a test Step that allows delegation to XCTAssert with access to the state (doesn't need to be XCTAssert, could just return true if step successful/false if not), is to allow fuzzy matching - ex. if you want to check that `alert` has some value, but you don't care exactly what value. This could allow for tests to be less brittle (exact wording of alert doesn't matter), or to delegate that precision to other tests (ex. one test already checks the exact wording, another test is checking something that requires the alert to be presented but doesn't care about the wording).

Reply:
Yeah, we would still like people to try to be as exhaustive as possible with their assertions, but I can definitely see a use case for something like that. It would be nice if the steps were more extensible than they are now. Something to think about for the future.